### PR TITLE
Added health information to top of screen

### PR DESCRIPTION
**Pull request by TimBeckmann**

### CODE REVIEW

File: src/output.cpp [hunk 1]
```diff
@@ -14,7 +15,23 @@ void clearScreen() {
     [[maybe_unused]] const auto test = system("clear");
 #endif
 }
+/*
+    Used to count the ships on both boards and displays them on the screen
+*/
+void healthBar(const BoardType& playerBoard, const BoardType& computerBoard) {
+    size_t playerHealth = 0, computerHealth = 0;
```
Comment (jotoho) [inline src/output.cpp:22]: `size_t` -> `std::size_t`

File: src/output.cpp [hunk 2]
```diff
@@ -14,7 +15,23 @@ void clearScreen() {
...
+    for (const auto& segment : playerBoard)
+        playerHealth +=
+            std::count(segment.begin(), segment.end(), FieldValue::SHIP);
+
+    for (const auto& segment : computerBoard)
+        computerHealth +=
+            std::count(segment.begin(), segment.end(), FieldValue::SHIP);
```
Comment (jotoho) [inline src/output.cpp:29]: Please extract this into it's own function. This kind of repeated code should be avoided.

File: src/output.cpp [hunk 3]
```diff
@@ -46,13 +63,19 @@ void writeProgramInfo(std::ostream& out) {
...
     Used to update the board on screen
 */
-void refreshBoard(const BoardType& board,
+void refreshBoard(const BoardType& playerBoard,
+                  const BoardType& computerBoard,
+                  const BoardType& currentBoard,
                   const char* const boardDiscription,
                   OutputTranslationTable translationTable) {
+    if (&currentBoard != &playerBoard || &currentBoard != &computerBoard) {
+        throw std::invalid_argument{"Boards are out of sync."};
```
Comment (jotoho) [inline src/output.cpp:72]: "Reference to wrong board was passed to refreshBoard"

**Review by jotoho — CHANGES_REQUESTED** [review]

File: src/output.cpp [hunk 4]
```diff
@@ -46,13 +68,20 @@ void writeProgramInfo(std::ostream& out) {
 /*
     Used to update the board on screen
 */
-void refreshBoard(const BoardType& board,
+void refreshBoard(const BoardType& playerBoard,
+                  const BoardType& computerBoard,
+                  const BoardType& currentBoard,
                   const char* const boardDiscription,
                   OutputTranslationTable translationTable) {
+    if (&currentBoard != &playerBoard || &currentBoard != &computerBoard) {
```
Comment (TimBeckmann) [inline src/output.cpp:76]: The part is throwing invalid arguments on game start @jotoho

Comment (jotoho) [reply src/output.cpp:76]: @TimBeckmann You're using the wrong logic operator. We need `&&`, not `||`

Comment (TimBeckmann) [reply src/output.cpp:76]: oh

File: src/output.cpp [hunk 5]
```diff
@@ -15,6 +16,27 @@ void clearScreen() {
...
+/*
+    Used to count the ships on both boards and displays them on the screen
+*/
+void healthBar(const BoardType& playerBoard, const BoardType& computerBoard) {
+    std::size_t playerHealth = 0, computerHealth = 0;
+
+    playerHealth = shipCount(playerBoard);
+    computerHealth = shipCount(computerBoard);
+    std::cout << std::setfill(' ') << std::setw(3) << "(Player 1) Computer "
+              << computerHealth << ':' << playerHealth
```
Comment (jotoho) [inline src/output.cpp:37]: Please add a space left and right of the `:`.

File: src/output.cpp [hunk 6]
```diff
@@ -15,6 +16,27 @@ void clearScreen() {
...
+    return ships;
+}
+/*
+    Used to count the ships on both boards and displays them on the screen
+*/
+void healthBar(const BoardType& playerBoard, const BoardType& computerBoard) {
+    std::size_t playerHealth = 0, computerHealth = 0;
+
+    playerHealth = shipCount(playerBoard);
+    computerHealth = shipCount(computerBoard);
```
Comment (jotoho) [inline src/output.cpp:35]: Refactor this to directly initialize the variables with the function output

**Review by jotoho — APPROVED** [review]

**Pull request merged**